### PR TITLE
Pulled taar_locale over from python_mozetl 

### PR DESCRIPTION
**Pull request by crankycoder**
* inlined taar_utils functions
* replaced s3_submission_date with the current clients_daily date
* removed temporary file requirement for the TAAR Locale
* updated boto code to write JSON directly into S3
* updated output S3 paths for taar_locale job

### CODE REVIEW

File: jobs/taar_locale.py [hunk 1]
```diff
@@ -0,0 +1,444 @@
...
+        AND addon_info.is_system = FALSE -- exclude system addons
+        AND locale <> 'null'
+        AND addon_info.addon_id IS NOT NULL
+    """
+
+    addons_query = addons_query_template.format(start_date=start_date)
+
+    # Could scale this up to grab more than what is in
+    # longitudinal and see how long it takes to run.
+    s3_url = "s3a://telemetry-parquet/clients_daily/v6/submission_date_s3={}".format(
```
Comment (crankycoder) [inline jobs/taar_locale.py]: @akkomar I am still using s3a URLs here - I'm not sure if I should be using `gs://` URLs here, but I can't seem to see a `telemetry-parquet` bucket in GCS.

**Comment by crankycoder** [comment]
This PR is going to have the same limited security issue as the previously migrated taar-lite jobs.  I'd like to get all the GCS->S3 syncing done in a separate PR.

**Comment by crankycoder** [comment]
@akkomar thanks.

I've updated the comment header to link the original source and test code in python_mozetl.
Commit message has been cleaned up and the parquet read is running off of GCS.

File: jobs/taar_locale.py [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+        AND addon_info.user_disabled = FALSE -- active addons only get counted
+        AND addon_info.app_disabled = FALSE -- exclude compatibility disabled addons
+        AND addon_info.is_system = FALSE -- exclude system addons
+        AND locale <> 'null'
+        AND addon_info.addon_id IS NOT NULL
+    """
+
+    addons_query = addons_query_template.format(start_date=start_date)
+
+    s3_url = "gs://moz-fx-data-derived-datasets-parquet/clients_daily/v6/submission_date_s3={}".format(
```
Comment (akkomar) [inline jobs/taar_locale.py]: This is no longer s3:
```suggestion
    gcs_uri = "gs://moz-fx-data-derived-datasets-parquet/clients_daily/v6/submission_date_s3={}".format(
```

Comment (crankycoder) [reply jobs/taar_locale.py]: I've fixed this to be gs_url now

**Pull request merged**